modules/core/src/main/java/org/apache/ignite/internal/managers/encryption/GridEncryptionManager.java

```diff
@@ -494,6 +547,56 @@ public void groupKey(int grpId, byte[] encGrpKey) {
         }
     }
 
+    /** {@inheritDoc} */
+    @Override public IgniteFuture<Void> changeMasterKey(String masterKeyName) {
+        if (ctx.clientNode())
+            throw new UnsupportedOperationException("Client and daemon nodes can not perform this operation.");
+
+        if (!ctx.state().clusterState().active())
+            throw new IgniteException("Master key change was rejected. The cluster is inactive.");
+
+        checkMasterKeyChangeSupported();
+
+        if (masterKeyName.equals(getMasterKeyName()))
+            throw new IgniteException("Master key change was rejected. Master key with provided name is already set.");
```

Comment:
> Master key change was rejected. Master key with provided name is already set.

Let's reformulate it "Master key change was rejected. New name equal to the current"

Reply:
Done.